### PR TITLE
Support Gradle 8

### PR DESCRIPTION
**Pull request by vierbergenlars**
In one go, also remove usage of conventions from the amp plugin (not from the alfresco plugin) to support
Gradle 9 which will get rid of Conventions completely

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
           fetch-depth: 0
       - uses: actions/setup-java@v1
         with:
-          java-version: 11
+          java-version: 17
```
Comment (WimCrols) [inline .github/workflows/ci.yml:20]: (non-blocking) Any reason for going to 17 if you still stay on 11 for the code itself?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+## Version 1.2.0 - 2024-08-07
+
+* [#100](https://github.com/xenit-eu/alfresco-gradle-sdk/pull/100) Support for Gradle 8
+
+> [!warning]
+> This release drops support for Gradle versions <7.1 and Java <11
```
Comment (WimCrols) [inline CHANGELOG.md:10]: Please update README too then. It still mentions Gradle 5.1.

File: settings.gradle [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+    id 'org.ajoberstar.reckon.settings' version "0.18.3"
+}
 rootProject.name = 'alfresco-sdk'
 
-enableFeaturePreview('STABLE_PUBLISHING')
\ No newline at end of file
+reckon {
+    defaultInferredScope = 'patch'
+    snapshots()
+    scopeCalc = calcScopeFromProp()
```
Comment (WimCrols) [inline settings.gradle:9]: I haven't used reckon myself, but if you use `calcScopeFromProp`, don't you need commit messages that follow the format outlined in https://github.com/ajoberstar/reckon?tab=readme-ov-file#reading-scope-from-commit-messages ? I don't see those in the history of this repo.

**Review by WimCrols — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vierbergenlars) [reply .github/workflows/ci.yml:20]: Yes; the sonarqube Gradle plugin requires Java 17 to work.

So the build itself needs to run with Java 17. The code itself is compiled with Java 11, so you can still run Gradle with an older Java version (e.g. for older Alfresco builds)

File: settings.gradle (hunk 3 above)
Comment (vierbergenlars) [reply settings.gradle:9]: That is for when you use `calcScopeFromCommitMessages()`, not `calcScopeFromProp()` (which reads from gradle properties that can be passed on the commandline)

**Review by WimCrols — APPROVED** [review]

**Pull request merged**